src/pages/MarketPlacePage/MarketPlaceTaskDetail.tsx

```diff
@@ -0,0 +1,74 @@
+import { useState, useEffect } from "react";
+import { useParams } from "react-router-dom";
+
+
+import projectData from "src/data/dummy_data_extended.json";
+import { slugify } from "@/src/utils/string-utils";
+
+import Tag from "@/src/components/Tag/Tag";
+import "./MarketPlaceTaskDetail.scss";
+import TaskPosterCard from "./components/TaskPosterCard/TaskPosterCard";
+
+function MarketplaceTaskDetail() {
+  const { id } = useParams();
+  const [jobs, setJobs] = useState([]);
+  const [jobDetail, setJobDetail] = useState();
+
+  useEffect(() => {
+    setJobs(projectData.jobs)
+
+    findJobDetail()
+  })
+  
+
+  const findJobDetail = () => {
```

Comment:
You will probably want to wrap this with a useCallback as the way to prevent rerenders.

You can also consider useMemo to prevent rerenders. Something I prefer to stop data from constantly having to reload:

```
  const jobDetail = useMemo(() => {
    return jobs.find((job) => slugify(job.name) === id);
  }, [id, jobs]);

  if (!jobDetail) {
    return <div>Loading...</div>;
  }
```